include/fiction/technology/charge_distribution_surface.hpp

```diff
@@ -1966,59 +1962,58 @@ class charge_distribution_surface<Lyt, false> : public Lyt
     }
     /**
      *  The stored unique index is converted to a charge distribution.
+     *
+     *  @param incremented Flag that can be set to true when the charge index was incremented before calling this
+     * function to enable an optimization. When set to true, the leading zeroes of the charge index are not handled.
      */
-    void index_to_charge_distribution() noexcept
+    void index_to_charge_distribution(const bool incremented = false) noexcept
```

Comment:
Instead of `bool` arguments, we usually prefer a dedicated `enum class` that provides more meaning to the arguments. From the users' perspective, there is not much meaning in a statement like `index_to_charge_distribution(false);`. However, `index_to_charge_distribution(charge_index_recomputation::FROM_SCRATCH);` (or similar) is way more intuitive.

Reply:
#347 implements this change